frontend/src/components/page_content/plan/debt.js

```diff
@@ -24,31 +26,55 @@ class Debt extends React.Component {
         fill: "var(--toast-neutral-3)",
       },
       {
-        name: "Target Savings",
+        name: "Max Savings",
         value: this.props.debtMonthly,
         fill: "url(#gradient)",
       },
     ];
+
+    this.actual = [
+      {
+        name: "Salary After Debt",
+        value:
+          calcMonthlyValue(this.props.salaryAfterTax) - this.props.loanDebt ||
+          0,
+        fill: "var(--toast-neutral-3)",
+      },
+      {
+        name: "Current Savings",
+        value: parseFloat(this.props.loanDebt || "0"),
+        fill: "url(#gradient)",
+      },
+    ];
+
+    this.setActiveToggle = this.setActiveToggle.bind(this);
+    this.getOnTrack = this.getOnTrack.bind(this);
+
+    this.currentDebt = React.createRef();
+    this.targetDebt = React.createRef();
   }
 
   getHeader() {
     return `$${numWithCommas(this.props.debtMonthly)}`;
   }
 
   getOnTrack() {
-    if (this.props.isOnTrackDebt === undefined) {
+    //let isOntrack = this.props.isOnTrackDebt; // Use backend
+    let isOntrack =
+      parseFloat(this.props.loanDebt || "0") <= this.props.debtMonthly; // Use calculated
+    if (isOntrack === undefined) {
       return "";
-    } else if (this.props.isOnTrackDebt) {
+    } else if (isOntrack) {
       return "You are currently on track.";
     } else {
-      return "You are currently not on track.";
+      return "You are currently not on track since you plan to repay debt that is more than 36% your monthly income.";
```

Comment:
i dont think we can hardcode the 36% still if the advisor can override the 36% rule?